sdk/keyvault/Azure.Security.KeyVault.Shared/src/ChallengeBasedAuthenticationPolicy.cs

```diff
@@ -63,190 +67,64 @@ protected async Task PreProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPip
+            internal AuthorityScope(string authrority, string[] scopes)
             {
-                AuthenticationChallenge challenge = null;
-
-                if (response.Headers.TryGetValue("WWW-Authenticate", out string challengeValue) && challengeValue.StartsWith(BearerChallengePrefix, StringComparison.OrdinalIgnoreCase))
-                {
-                    challenge = ParseBearerChallengeHeaderValue(challengeValue);
-                }
-
-                return challenge;
+                Authority = authrority;
+                Scopes = scopes;
             }
+            public string Authority { get; }
 
-            private static AuthenticationChallenge ParseBearerChallengeHeaderValue(string challengeValue)
-            {
-                string authority = null;
-                string scope = null;
-
-                // remove the bearer challenge prefix
-                var trimmedChallenge = challengeValue.Substring(BearerChallengePrefix.Length);
-
-                // Split the trimmed challenge into a set of name=value strings that
-                // are comma separated. The value fields are expected to be within
-                // quotation characters that are stripped here.
-                string[] pairs = trimmedChallenge.Split(s_challengeDelimiters, StringSplitOptions.RemoveEmptyEntries);
+            public string[] Scopes { get; }
+        }
 
-                if (pairs.Length > 0)
-                {
-                    // Process the name=value string
-                    for (int i = 0; i < pairs.Length; i++)
-                    {
-                        string[] pair = pairs[i].Split('=');
-
-                        if (pair.Length == 2)
-                        {
-                            // We have a key and a value, now need to trim and decode
-                            string key = pair[0].AsSpan().Trim().Trim('\"').ToString();
-                            string value = pair[1].AsSpan().Trim().Trim('\"').ToString();
-
-                            if (!string.IsNullOrEmpty(key))
-                            {
-                                // Ordered by current likelihood.
-                                if (string.Equals(key, "authorization", StringComparison.OrdinalIgnoreCase))
-                                {
-                                    authority = value;
-                                }
-                                else if (string.Equals(key, "resource", StringComparison.OrdinalIgnoreCase))
-                                {
-                                    scope = value + "/.default";
-                                }
-                                else if (string.Equals(key, "scope", StringComparison.OrdinalIgnoreCase))
-                                {
-                                    scope = value;
-                                }
-                                else if (string.Equals(key, "authorization_uri", StringComparison.OrdinalIgnoreCase))
```

Comment:
Does the base implementation handle this token? Rare, but present in older APIs. @schaabs, is this even worth worrying about in track 2? This was ported from track 1 originally.